src/Integration.Vsix/SLCore/RaiseIssueParamsToAnalysisIssueConverter.cs

```diff
@@ -0,0 +1,112 @@
+    public class RaiseIssueParamsToAnalysisIssueConverter : IRaiseIssueParamsToAnalysisIssueConverter
+    {
+        private readonly ITextDocumentFactoryService textDocumentFactoryService;
+        private readonly IContentTypeRegistryService contentTypeRegistryService;
+        private readonly ILineHashCalculator lineHashCalculator;
+
+        [ImportingConstructor]
+        public RaiseIssueParamsToAnalysisIssueConverter(ITextDocumentFactoryService textDocumentFactoryService, IContentTypeRegistryService contentTypeRegistryService)
+            : this(textDocumentFactoryService, contentTypeRegistryService, new LineHashCalculator())
+        { }
+
+        internal RaiseIssueParamsToAnalysisIssueConverter(ITextDocumentFactoryService textDocumentFactoryService, IContentTypeRegistryService contentTypeRegistryService, ILineHashCalculator lineHashCalculator)
+        {
+            this.textDocumentFactoryService = textDocumentFactoryService;
+            this.contentTypeRegistryService = contentTypeRegistryService;
+            this.lineHashCalculator = lineHashCalculator;
+        }
+
+        public IEnumerable<IAnalysisIssue> GetAnalysisIssues(RaiseIssuesParams raiseIssuesParams)
+        {
+            var result = new List<IAnalysisIssue>();
+            foreach (var issue in raiseIssuesParams.issuesByFileUri)
+            {
+                foreach (var item in issue.Value)
+                {
+                    result.Add(new AnalysisIssue
+                        (item.ruleKey,
+                        item.severity.ToAnalysisIssueSeverity(),
+                        item.type.ToAnalysisIssueType(),
+                        GetHighestSoftwareQualitySeverity(item.impacts),
+                        GetAnalysisIssueLocation(issue.Key.LocalPath, item.primaryMessage, item.textRange),
+                        item.flows?.Select(GetAnalysisIssueFlow).ToList(),
+                        item.quickFixes?.Select(qf => GetQuickFix(issue.Key, qf)).Where(qf => qf is not null).ToList(),
+                        item.ruleDescriptionContextKey
+                        ));
+                }
+            }
+
+            return result;
+        }
+
+        private SoftwareQualitySeverity? GetHighestSoftwareQualitySeverity(IEnumerable<ImpactDto> impacts)
+        {
+            if (impacts is not null && impacts.Any())
+            {
+                return impacts.Max(i => i.impactSeverity).ToSoftwareQualitySeverity();
+            }
+            return null;
+        }
+
+        private IAnalysisIssueLocation GetAnalysisIssueLocation(string filePath, string message, TextRangeDto textRangeDto)
+        {
+            var textDocument = textDocumentFactoryService.CreateAndLoadTextDocument(filePath, contentTypeRegistryService.UnknownContentType); // load the document from disc
+            var currentSnapshot = textDocument.TextBuffer.CurrentSnapshot;
+
+            var lineHash = lineHashCalculator.Calculate(currentSnapshot, textRangeDto.startLine + 1);
```

Comment:
This is incorrect. The range hash should come from Sloop. If it doesn't, we should use ranges without the hash.

Range hash calculation is also done incorrectly here. You're calculating based on current snapshot, not based on the snapshot at the time of the analysis. This was part of the reason to introduce IssueConsumerStorage, to map to correct snapshot and translate to current.

So, remove references to VS-aware interfaces, simplify this class + tests and move it to the SLCore assembly.

Reply:
This the logic we currently use, I have added this so we can have the logic since we can get it from SLOOP.

Reply:
Can get it from sloop or cannot?

Also, it doesn't matter that it's the current logic. It's incorrect. You're calculating line hashes based on a potentially different snapshot, without translating the position. What you end up with is potentially incorrect line hashes that can lead to false-negative location matching. Don't make your life harder by writing extra code that doesn't even work properly.

Reply:
https://github.com/SonarSource/sonarlint-visualstudio/blob/6c0113fa0422aa08edd71533dcfe3ea162701b84/src/CloudSecrets/SecretsAnalyzer.cs#L102

Reply:
It's not even the same scenario. There the content is used for analysis